spec/lib/telephony/otp_sender_spec.rb

```diff
@@ -255,4 +255,100 @@
       end
     end
   end
+
+  describe '#authentication_message' do
+    let(:sender) do
+      sender = Telephony::OtpSender.new(
+        to: '+18888675309',
+        otp: 'ABC123',
+        channel: 'sms',
+        expiration: TwoFactorAuthenticatable::DIRECT_OTP_VALID_FOR_MINUTES,
+        domain: 'secure.login.gov',
+        country_code: 'US',
+      )
+    end
+
+    context 'sms' do
+      context 'English' do
+        it 'does not contain any non-GSM characters and is less than or equal to 160 characters' do
+          message = sender.authentication_message
+          expect(Telephony.sms_parts(message)).to eq 1
+          expect(message.chars.all? { |x| Telephony::GSM_CHARACTERS.include?(x) }).to eq true
+        end
+      end
+
+      # The Spanish-language translation currently includes the 'ó', character, which is not
+      # in the GSM 03.38 character set
+      context 'Spanish' do
+        it 'is sent in three parts' do
+          I18n.locale = :es
+
+          message = sender.authentication_message
+          expect(Telephony.sms_parts(message)).to eq 3
+        ensure
+          I18n.locale = :en
+        end
+      end
+
+      context 'French' do
+        it 'does not contain any non-GSM characters and is less than or equal to 160 characters' do
+          I18n.locale = :fr
+
+          message = sender.authentication_message
+          expect(Telephony.sms_parts(message)).to eq 1
+          expect(message.chars.all? { |x| Telephony::GSM_CHARACTERS.include?(x) }).to eq true
+        ensure
+          I18n.locale = :en
+        end
```

Comment:
for all these specs with a locale, what if we used a `let` var + an around filter at the top of the file?

```
let(:locale) { :es }
around do |ex|
  I18n.locale = local
  ex.run
ensure
  I18n.locale = :en
end
```

Then we can clean up all these separate `ensure` blocks?

Reply:
I wasn't able to get this working, I tried:

```diff
+    around do |ex|
+      puts 'around block'
+      puts locale
+      I18n.locale = locale
+      puts I18n.locale
+      puts 'starting spec'
+      ex.run
+    ensure
+      I18n.locale = :en
+    end
...
+       let(:locale) { :es }
```

But the logs show the locale and locale variable being set correctly in the block, but the locale is reset in the actual spec?

```
around block
es
es
starting spec
in spec
es
en
```

Reply:
Oh I think it must be related to this: https://github.com/18F/identity-idp/blob/64be38f88429503af7d258ba3e422daa6216e3e5/spec/rails_helper.rb#L68-L70

so I think `around` gets called and that `before` block is called before the example

and in other specs we only set locale with a `before` block

```
before { I18n.locale = :es }
```

Reply:
I'd love to clean up these repeated ensure clauses somehow, but I wouldn't consider it a blocker to merging this PR

Reply:
Doesn't RSpec have an `after` method that can be used?

Reply:
I tested and the following works with dropping the `ensure` blocks:

```diff
+      after do
+        I18n.locale = :en
+      end
+
       context 'English' do
```

but the net number of lines is the same, and moves the context further away from why it's being reset so I'm leaning towards keeping as-is for now

Reply:
makes sense to me!